### PR TITLE
METRON-1196 Increment master version number to 0.4.2 for on-going development

### PR DESCRIPTION
**Pull request by mattf-horton**
## Contributor Comments
Standard update of the Metron version number.
There are just a couple files where I changed a 0.4.0 to 0.4.1 to make a multi-line example self-consistent, that had been poorly edited in the past (probably by me).  The rest of it is all changing "0.4.1" to "0.4.2" where needed.

If you grep for "0.4.1" (or rather for "0\\.4\\.1"), you'll find some remaining usage of 0.4.1 in documentation files, where I judged the usefulness of an example did not require updating the version number.  Also of course "0.4.1" persists in history and log files, and in usages related to package versions other than metron.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [NA] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [NA] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [NA] Have you written or updated unit tests and or integration tests to verify your changes?
- [NA] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [NA] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [NA] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

### CODE REVIEW

**Comment by kylerichardson** [comment]
There are so many `pom.xml` files to change. Have we considered using a metron_version property in the parent/root pom to avoid needing to change so many files in future releases? If folks agree, I'd be happy to do the leg work and submit a PR.

**Comment by mattf-horton** [comment]
Sure, but can we do that as a separate Jira?  We should advance the present version number promptly, since release 0.4.1 is published.

**Comment by kylerichardson** [comment]
Absolutely, I've logged METRON-1210 for this.

File: metron-platform/README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ limitations under the License.
 
 # Current Build
 
-The latest build of metron-platform is 0.4.1.
+The latest build of metron-platform is 0.4.2.
```
Comment (ottobackwards) [inline metron-platform/README.md]: Is this statement still valid?  Just as an aside

**Comment by ottobackwards** [comment]
```bash
metron-analytics/metron-profiler-client/README.md:364:[root@node1 0.4.1]# bin/stellar -z node1:2181

metron-analytics/metron-profiler/README.md:26:    metron-deployment//packaging/docker/rpm-docker/RPMS/noarch/metron-profiler-0.4.1-201707131420.noarch.rpm
metron-analytics/metron-profiler/README.md:35:    /localrepo/metron-profiler-0.4.1-201707112313.noarch.rpm
metron-analytics/metron-profiler/README.md:49:    /usr/metron/0.4.1
metron-analytics/metron-profiler/README.md:50:    /usr/metron/0.4.1/bin
metron-analytics/metron-profiler/README.md:51:    /usr/metron/0.4.1/bin/start_profiler_topology.sh
metron-analytics/metron-profiler/README.md:52:    /usr/metron/0.4.1/config
metron-analytics/metron-profiler/README.md:53:    /usr/metron/0.4.1/config/profiler.properties
metron-analytics/metron-profiler/README.md:54:    /usr/metron/0.4.1/flux
metron-analytics/metron-profiler/README.md:55:    /usr/metron/0.4.1/flux/profiler
metron-analytics/metron-profiler/README.md:56:    /usr/metron/0.4.1/flux/profiler/remote.yaml
metron-analytics/metron-profiler/README.md:57:    /usr/metron/0.4.1/lib
metron-analytics/metron-profiler/README.md:58:    /usr/metron/0.4.1/lib/metron-profiler-0.4.1-uber.jar


```

**Comment by justinleet** [comment]
@mattf-horton @ottobackwards Is there any more work for this?

**Comment by ottobackwards** [comment]
I don't see any commits since my comments, so that still needs to be addressed I think

File: metron-platform/README.md (hunk 1 above)
Comment (mattf-horton) [reply metron-platform/README.md]: The above statement is validly updated to reflect the new upcoming version, which is the point of this PR.  However, the file metron-platform/README.md as a whole doesn't make much sense any more.  I've opened https://issues.apache.org/jira/browse/METRON-1271 to address it.

**Comment by mattf-horton** [comment]
@ottobackwards , regarding the version references in metron-analytics/metron-profiler/README.md:
This is a little difficult.  Although these are installation instructions, the lines you grepped are all OUTPUT from `find` or `rpm`.  In use with an 0.4.2 code base, the actual output will be the correct-in-that-context 0.4.2 filenames and pathnames.  So while user might be surprised at first, they will not (if they're following instructions) be caused to type incorrect commands.  And since 0.4.2 hasn't been released yet, there is no date field that I can substitute for `201707131420` would be "correct" with 0.4.2.

I think it's best to leave it the way it is.  If you prefer a different rendering, please propose.

Regarding the line from metron-analytics/metron-profiler-client/README.md that has an 0.4.1, this is in the text of an example, and does not materially affect the example.  Therefore, as noted in the opening comments of this PR, it is my policy to leave it unchanged.  "If you grep for `0.4.1` (or rather for `0\\.4\\.1`), you'll find some remaining usage of 0.4.1 in documentation files, where I judged the usefulness of an example did not require updating the version number."

Open to discussion, of course.  But by following this policy, those (many) lines of text fall out of the grep on the next generation (when moving from 0.4.2 to 0.4.3).  Which makes release management much easier.

**Comment by ottobackwards** [comment]
I don't have strong enough feelings to hold this back.  I think that maybe we should make an effort to use $METRON_VERSION or something in the readme's etc.  But that is a different pr etc.

**Comment by mattf-horton** [comment]
@ottobackwards , yup.  So is that a +1 regarding commit? :-)

**Comment by ottobackwards** [comment]
+1

**Comment by mattf-horton** [comment]
Thanks, @ottobackwards and @kylerichardson .  Committing...

**Comment by ottobackwards** [comment]
So, between the original scan and the commit, metron-platform/metron-zookeeper was added.  It is not addressed in this.  I'm trying to get a PR updated with this, but cannot get metron-test-utilites to build.

We may need a quick fix, unfortunately I will not be able to for a few hours, maybe someone else can grab it?

**Comment by cestella** [comment]
Yeah, I got it.  https://issues.apache.org/jira/browse/METRON-1280

**Comment by ottobackwards** [comment]
Ok, Master is broken in travis as well

**Comment by mattf-horton** [comment]
Sorry about that, folks.  In future I won't trust github when it says it will still merge "without conflict".

**Comment by mattf-horton** [comment]
Needed to re-kick Travis to be sure.

**Pull request closed without merging**